apps/smtp/src/modules/smtp/smtp-email-sender.ts

```diff
@@ -0,0 +1,103 @@
+import nodemailer from "nodemailer";
+import { createLogger } from "../../logger";
+import { SmtpEncryptionType } from "./configuration/smtp-config-schema";
+
+export interface SendMailArgs {
+  smtpSettings: {
+    host: string;
+    port: number;
+    auth?: {
+      user: string;
+      pass: string | undefined;
+    };
+    encryption: SmtpEncryptionType;
+  };
+  mailData: {
+    from: string;
+    to: string;
+    text: string;
+    html: string;
+    subject: string;
+  };
+}
+
+export interface ISMTPEmailSender {
+  sendEmailWithSmtp({ smtpSettings, mailData }: SendMailArgs): Promise<{ response: any }>;
+}
+
+export class SmtpEmailSender implements ISMTPEmailSender {
+  private logger = createLogger("SmtpEmailSender");
+
+  async sendEmailWithSmtp({ smtpSettings, mailData }: SendMailArgs) {
+    this.logger.debug("Sending an email with SMTP");
+
+    let transporter: nodemailer.Transporter;
+
+    /*
+     * https://github.com/nodemailer/nodemailer/issues/1461#issuecomment-1263131029
+     * [secure argument] it’s not about security but if the server starts tcp connections over TLS mode or not.
+     * If it starts connections in cleartext mode, the client can not use TLS until STARTTLS can be established later.
+     */
+
+    switch (smtpSettings.encryption) {
+      case "TLS":
+        transporter = nodemailer.createTransport({
+          tls: {
+            minVersion: "TLSv1.1",
```

Comment:
question: shouldn't we use TLSv1.2?

Reply:
I will check this later, its copy pasted